src/main/java/com/limechain/network/protocol/blockannounce/BlockAnnounceEngine.java

```diff
@@ -91,6 +98,19 @@ private void handleBlockAnnounce(byte[] msg, PeerId peerId) {
         if (BlockState.getInstance().isInitialized()) {
             BlockState.getInstance().addBlockToBlockTree(announce.getHeader());
         }
+        updateEpochStateIfBabeConsensusMessageExists(announce);
```

Comment:
So basically I think in future we need some **BabeSync class (or BabeHelper)** that could invoke this code, meaning that BlockAnnounceEngine would be more single responsible. Also we can put this logic inside of **warpSyncState.syncBlockAnnounce** . Or it could be some **DigestHelper**  that would have all logic needed for all types message handling. For now I put logic for consensus message into method below. What do you think?